flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/partitioner/ShufflePartitionerTest.java

```diff
@@ -30,35 +26,23 @@
 /**
  * Tests for {@link ShufflePartitioner}.
  */
-public class ShufflePartitionerTest {
-
-	private ShufflePartitioner<Tuple> shufflePartitioner;
-	private StreamRecord<Tuple> streamRecord = new StreamRecord<Tuple>(null);
-	private SerializationDelegate<StreamRecord<Tuple>> sd = new SerializationDelegate<StreamRecord<Tuple>>(
-			null);
+public class ShufflePartitionerTest extends StreamPartitionerTest {
 
 	@Before
 	public void setPartitioner() {
-		shufflePartitioner = new ShufflePartitioner<Tuple>();
-	}
-
-	@Test
-	public void testSelectChannelsLength() {
-		sd.setInstance(streamRecord);
-		assertEquals(1, shufflePartitioner.selectChannels(sd, 1).length);
-		assertEquals(1, shufflePartitioner.selectChannels(sd, 2).length);
-		assertEquals(1, shufflePartitioner.selectChannels(sd, 1024).length);
+		streamPartitioner = new ShufflePartitioner<>();
```

Comment:
If you define abstract method `StreamPartitioner<Tuple> createPartitioner()` in `StreamParitionerTest` then if someone forgets to override it he will get a compilation error instead of `NullPointerException` as he would now (if he would forgot to declare `@Before` method).

Reply:
make sense!